packages/truffle-compile/compilerSupplier/index.js

```diff
@@ -13,9 +13,19 @@ const {
 class CompilerSupplier {
   constructor(_config) {
     _config = _config || {};
-    const defaultConfig = { version: null };
+    const defaultConfig = {
+      version: null,
+      compilerRoots: [
+        "https://solc-bin.ethereum.org/bin/",
+        "https://ethereum.github.io/solc-bin/bin/",
+        "https://relay.trufflesuite.com/solc/bin/"
+      ]
+    };
     this.config = Object.assign({}, defaultConfig, _config);
-    this.strategyOptions = { version: this.config.version };
+    this.strategyOptions = {
+      version: this.config.version,
+      compilerRoots: this.config.compilerRoots
+    };
```

Comment:
Adding this here now suggests to me that maybe we should move all of the `compilerRoots` stuff to VersionRange, since that's the only place where it's used.

Reply:
`compilerRoots` is configurable by the user to I need to pass the user config to the `VersionRange`. 
I have removed the default `compilerRoots` from `compilerSupplier`.